### PR TITLE
Add github action for python3.10b4

### PR DESCRIPTION
**Pull request by honglei**
[Failed to create a python 3.10.0-beta.4 env in github action ](https://github.com/tox-dev/tox/issues/2109)

https://github.com/ymyzk/tox-gh-actions

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -21,6 +21,8 @@ jobs:
             postgres: 12
           - python: 3.9
             postgres: 13
+          - python: 3.10.0-beta.4
```
Comment (hugovk) [inline .github/workflows/tests.yml:24]: At least for GHA, this is simpler and will pick up the latest, for example it's now RC1:

```suggestion
          - python: 3.10-dev
```

**Comment by dvarrazzo** [comment]
Can appveyor be fixed?

**Comment by honglei** [comment]
@dvarrazzo I'm not familar with appveyor, but I already see that pyzmq and asyncpg already provide wheel for Python3.10,
and these maybe a good example for publish wheels to pypi automatically.
https://github.com/zeromq/pyzmq/blob/main/.github/workflows/wheels.yml

https://github.com/MagicStack/asyncpg/blob/master/.github/workflows/release.yml 

```
  Using cached pyzmq-22.2.1-cp310-cp310-win_amd64.whl (1.1 MB)
Requirement already satisfied: sqlalchemy in c:\python310\lib\site-packages (from -r .\requirements.txt (line 17)) (1.4.23)
Collecting asyncpg
  Using cached asyncpg-0.24.0-cp310-cp310-win_amd64.whl (1.1 MB)
```

**Comment by dvarrazzo** [comment]
I am not interested in publishing wheels automatically: it is important enough that I prefer to do it manually, thank you.

I see that you have changed appveyor, and as a consequence the test fails, so this branch is not ready to be merged. Can you 1) fix appveyor or 2) avoid to propose a breaking change? 1) would be largely preferred.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -56,7 +58,7 @@ jobs:
         with:
           python-version: ${{ matrix.python }}
       - name: Install tox
-        run: pip install tox
+        run: pip install tox tox-gh-actions
```
Comment (dvarrazzo) [inline .github/workflows/tests.yml:61]: Why is this needed?

**Comment by cclauss** [comment]
Perhaps close or update for Python 3.11 beta 3?

**Comment by hugovk** [comment]
Can close, 3.11 was added in https://github.com/psycopg/psycopg2/commit/5fb1305a1420ecb884e581d84ea520e490dd5fdf
